### PR TITLE
Jakubsmahovsky/sc 1680/update cardano backend to current emurgo

### PR DESCRIPTION
**Pull request by JakubSmahovsky**
https://app.shortcut.com/dcspark/story/1680/update-cardano-backend-to-current-emurgo-version

**I need feedback:**
https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/305
prettier
This one is obviously very awkward, because it just creates a huge amount of conflicts. I took all our changes except for the package.json and pr-checks.yml and ran prettier on the result. Then I ran tests to check if something broke.

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/280
*"This PR changes the txs/signed endpoint to post to Yoroi queue instead of posting to the TX submission API directly."*
There is a config option to turn it on and it's off by defualt. I added this, it shouldn't break anything.

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/306
release 2.1.0
Added as an empty merge, because we're at 2.2.0.

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/286
refactor(cip12), refactors endpoint added in https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/248 , which we skipped
Skipped again, but we might want to add things like this in the future just to make it easier to sync with Emurgo.

**this PR is here ATM**

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/285
*"This is part of the Catalyst project funded effort to provide access to metadata oracles datapoints in Yoroi."*
TLDR: new endpoints + tests
Don't know

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/309
*"patch deserialization error for duplicate metadata map keys"*
Add some try-catch in case some deserialization fails.
I'll add this.

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/310/files
*"Serlib update + Prettier"*
I'll add this the same way as prettier: update the packages and run prettier myself.

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/311
release 2.2.0
Will add as an empty merge, because we're already at 2.2.0.

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/313
I don't know what this is

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/308
my PR, data_hash

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/314
*"this PR adds a new endpoint to check if given policyIds already exist on chain"*
TLDR: new endpoint, no tests

### CODE REVIEW

**Comment by SebastienGllmt** [comment]
Emurgo#285 -> ignore
Emurgo#313 -> add

Everything else you said LGTM

**Comment by JakubSmahovsky** [comment]
**IMPORTANT**:
1. I couldn't test this. As far as I know, I'm running tests on the currently deployed db-sync. It has `TransactionOutput` view already created and it doesn't want to drop it, so `txDataHash` is missing from it and because of that all tests using transactionHistory are failing.

2. I merged master instead of rebasing because rebasing generates a huge number of conflicts (because of prettier)

**Comment by JakubSmahovsky** [comment]
Adding newer PRs:

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/316
revert tx history change, improve error logging

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/317
fix data_hash

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/318
add slot number to best block endpoint

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/319
fix txStatus 404 error

https://github.com/Emurgo/yoroi-graphql-migration-backend/pull/322
fix dockerfile
this is a revert of an change earlier in this sync

**Review by SebastienGllmt — APPROVED** [review]
